### DISCUSSION TITLE
Support @NonNull at package-info.java level

### DISCUSSION THREAD

**Discussion by cykl** [Randgalt/record-builder#89] [Ideas]
A common way to manage nullability annotations is to put a default @NonNull annotation at method / parameter / field level using package-info.java and then only add @Nullable where relevant. Rational being that it is far more common to want something @NonNull than @Nullable.

 The idiomatic way to do it is to define an annotation like that

```java
@Target( { ElementType.PACKAGE, ElementType.TYPE } )
@Retention( RetentionPolicy.RUNTIME )
@Documented
@Nonnull
@TypeQualifierDefault( ElementType.FIELD )
public @interface NonNullFields {
}
```

And then apply it to the package-info.java:

```java
@NonNullApi
@NonNullFields
package my.package;
``` 

It tried to define a such annotation for record components and apply it. But record-builder seems to not care. 

```java
@Target( { ElementType.PACKAGE, ElementType.TYPE } )
@Retention( RetentionPolicy.RUNTIME )
@Documented
@Nonnull
@TypeQualifierDefault( { ElementType.RECORD_COMPONENT } )
public @interface NonNullRecordComponents {
} 
```

I have not yet dug into record-builder internals; is there a chance to be able to use this pattern with record-builder?

**Comment by Randgalt** [comment]
yeah - I'm OK with that. In fact, I thought that I'd done that already. But, a PR (or new issue) would be appreciated.
